Tchap/Modules/Favourites/FavouriteMessages/FavouriteMessagesViewModel.swift

```diff
@@ -155,47 +155,41 @@ final class FavouriteMessagesViewModel: NSObject, FavouriteMessagesViewModelType
                     }
                     
                     //  handle encryption for this event
-                    if event.isEncrypted && event.clear == nil {
-                        if self.session.decryptEvent(event, inTimeline: nil) == false {
-                            print("[FavouriteMessagesViewModel] processEditEvent: Fail to decrypt event: \(event.eventId ?? "")")
-                        }
+                    if event.isEncrypted && event.clear == nil && self.session.decryptEvent(event, inTimeline: nil) == false {
+                        print("[FavouriteMessagesViewModel] processEditEvent: Fail to decrypt event: \(event.eventId ?? "")")
                     }
                     
                     // Check whether the user knows this room to create the room data source if it doesn't exist.
                     roomDataSourceManager?.roomDataSource(forRoom: favouriteEvent.roomId, create: (self.session.room(withRoomId: favouriteEvent.roomId) != nil), onComplete: { roomDataSource in
                         
                         if roomDataSource != nil, let cellData = FavouriteMessagesBubbleCellData(event: event, andRoomState: roomDataSource?.roomState, andRoomDataSource: roomDataSource) {
-                            self.process(cellData: cellData)
+                            favouriteMessagesCache.append(cellData)
+                        }
+                        
+                        if completeOperations == limit {
+                            self.process(cellDatas: favouriteMessagesCache)
+                        } else {
+                            completeOperations += 1
```

Comment:
you missed some cases where decrease completeOperations, let's have a talk about this